src/nvim/api/buffer.c

```diff
@@ -493,11 +493,10 @@ void nvim_buf_set_lines(uint64_t channel_id,
 
 /// Return the byte offset for a line.
 //
-/// This includes the end-of-line character, depending on the 'fileformat'
-/// option for the current buffer. The first line returns 0. UTF-8 encoding is
-/// used, 'fileencoding' is ignored. Sending in the index just below the last
-/// line gives the total byte count for the entire file. A final newline is
-/// included if it would be written, see 'eol'.
+/// The first line returns 0. UTF-8 bytes are counted, and newline counts as
+/// one byte. 'fileformat' and 'fileencoding' are ignored. Sending in the
```

Comment:
This might be worth mentioning IIUC:

    EOL counts as one byte, regardless of 'fileformat'.

But in a unix-formatted file, will CRLF count as 2 bytes (because the CR is not recognized as a line-ending)?

Reply:
Is "newline counts as one byte. 'fileformat' and 'fileencoding' are ignored" clear enough?

Reply:
I think it's a bit ambiguous-sounding for most people who aren't used to thinking about details of line-endings. See my latest edit, I think it's worth explicitly saying what happens when fileformat is dos vs unix (which is counterintuitive to "ignored": the result changes when 'fileformat' changes!)

Reply:
Perhaps we should always count CRLF as 2 bytes, regardless of 'fileformat'. Or is that what already happens?

Reply:
With this change, the function _unambiguously_ and always represents the actual state of the buffer, independently of the value of `fileformat`, or `fileencoding`, or any arbitrary transformation that `charconvert` or `BufWriteCmd` are free to do between the internal state and the written bytes on the disk.

"EOL counts as one byte, thus CRLF counts as 1 byte if fileformat=dos and 2 bytes if fileformat=unix." is misleading and irrelevant, because changing `fileformat` will never change change how this function counts bytes in a buffer. It changes what buffer contents a `:read` operation results in, and it changes what `:write` writes disk from a given buffer (together with a lot of other factors), it doesn't change how this function behaves.